test/contact_properties.js

```diff
@@ -1,164 +1,296 @@
+    const updateEndpoint = {
+      path: `/properties/v1/contacts/properties/named/${properties.name}`,
+      response: { ...properties, description },
+    }
+    fakeHubspotApi.setupServer({ putEndpoints: [updateEndpoint] })
 
-    before(() => {
-      return hubspot.contacts.properties.get().then(results => {
-        // console.log(results)
-        propertyName = results[0].name
-      })
+    before(function() {
+      if (process.env.NOCK_OFF) {
+        return createTestProperty(properties)
+      }
+    })
+    after(function() {
+      if (process.env.NOCK_OFF) {
+        return deleteTestProperty(properties.name)
+      }
     })
 
-    it('should get a property by name', function() {
+    it('should update the property', function() {
       return hubspot.contacts.properties
-        .getByName(propertyName)
-        .then(results => {
-          // console.log(results)
-          expect(results).to.be.an('object')
-          expect(results).to.have.a.property('name')
+        .update(properties.name, { ...properties, description })
+        .then(data => {
+          expect(data.description).to.eq(description)
         })
     })
   })
 
   describe('upsert', function() {
-    it('should create or update the property', function() {
-      return hubspot.contacts.properties.upsert(property).then(data => {
-        expect(data).to.be.an('object')
-        expect(data).to.have.a.property('name')
+    describe("when the record doesn't exists", function() {
+      const createEndpoint = {
+        path: '/properties/v1/contacts/properties',
+        response: properties,
+      }
+      fakeHubspotApi.setupServer({ postEndpoints: [createEndpoint] })
+
+      after(function() {
+        if (process.env.NOCK_OFF) {
+          return deleteTestProperty(properties.name)
+        }
       })
-    })
-  })
 
-  describe('delete', function() {
-    const testDeleteProperty = {
-      name: 'delete_test_property_' + Date.now(),
-      label: 'node-hubspot test property',
-      groupName: 'contactinformation',
-      description: 'Test property',
-      type: 'string',
-      fieldType: 'text',
-      formField: false,
-      displayOrder: -1,
-      options: [],
-    }
-    it('can delete', function() {
-      return hubspot.contacts.properties
-        .upsert(testDeleteProperty)
-        .then(data => {
-          expect(data).to.be.an('object')
-          expect(data).to.have.a.property('name')
-          return hubspot.contacts.properties.delete(testDeleteProperty.name)
+      it('should create the property', function() {
+        return hubspot.contacts.properties.upsert(properties).then(data => {
+          expect(data.description).to.eq(properties.description)
         })
+      })
+    })
+
+    describe('when the record already exists', function() {
+      const description = 'Updated display name'
+      const createEndpoint = {
+        path: '/properties/v1/contacts/properties',
+        statusCode: 409,
```

Comment:
> The HTTP 409 Conflict response status code indicates a request conflict with the current state of the server.

TIL 🌈

Reply:
I think I would prefer to remove this functionality but since we already have a big breaking change I didn't want to introduce another. This upsert surprised me.